### DISCUSSION TITLE
Conpiling error

### DISCUSSION THREAD

**Discussion by ilkant** [Stellarium/stellarium#3956] [General]
First time I got an error compiling Stellarium on Kubuntu 24.10 system. I did
$ git pull https://github.com/Stellarium/stellarium
$ cmake -Who-dev -DCMAKE_INSTALL_PREFIX=/opt/stellarium ../.. 
 
compiling make -j12
...
[ 80%] Built target TelescopeControl_gui
[ 80%] Linking CXX static library libCalendars.a
[ 80%] Built target Calendars-static
make: *** [Makefile:156: all] Error 2
$

**Comment by 10110111** [comment]
You skipped the actual error in your build log.

**Reply by ilkant** [reply]
$ make -j12 >see.txt
/usr/bin/ld: ../libcommon.a(Spectrum.cpp.o): warning: relocation against `_ZTISt12out_of_range@@GLIBCXX_3.4' in read-only section `.text.unlikely'
/usr/bin/ld: ../libcommon.a(EclipsedDoubleScatteringPrecomputer.cpp.o): relocation R_X86_64_PC32 against symbol `_ZTISt12out_of_range@@GLIBCXX_3.4' can not be used when making a shared object; käännetty uudelleen -fPIC-parametrillä
/usr/bin/ld: loppulinkitys ei onnistunut: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [_deps/showmysky-qt6-build/ShowMySky/CMakeFiles/ShowMySky.dir/build.make:160: _deps/showmysky-qt6-build/ShowMySky/libShowMySky-Qt6.so.15.0.0] Virhe 1
make[1]: *** [CMakeFiles/Makefile2:3376: _deps/showmysky-qt6-build/ShowMySky/CMakeFiles/ShowMySky.dir/all] Virhe 2
make[1]: *** Odotetaan keskeneräisiä töitä....
make: *** [Makefile:156: all] Virhe 2

[see.txt](https://github.com/user-attachments/files/17444599/see.txt)

**Reply by 10110111** [reply]
Try this patch:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac653d5caa..37fd0327c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -691,8 +691,7 @@ SET(CMAKE_INCLUDE_CURRENT_DIR ON)
 IF(ENABLE_SHOWMYSKY)
     ADD_DEFINITIONS(-DENABLE_SHOWMYSKY)
     CPMFindPackage(NAME ShowMySky-Qt${QT_VERSION_MAJOR}
-        URL https://github.com/10110111/CalcMySky/archive/refs/tags/v0.3.3.tar.gz
-        URL_HASH SHA256=21cce3187009ce62a4a08a72b4e22ae7bc00654edfbc7fa9fe8ab65b412791df
+        URL https://codeload.github.com/10110111/CalcMySky/zip/refs/heads/master
         EXCLUDE_FROM_ALL yes
         OPTIONS "QT_VERSION ${QT_VERSION_MAJOR}")
     IF(ShowMySky-Qt${QT_VERSION_MAJOR}_ADDED)
```

**Reply by ilkant** [reply]
Command diff does not have --git parameter.

Then I did command:
$ git diff -- a/CMakelists.txt b/CMakeLists.txt index ac653d5caa..37fd0327c6 100644

It didn't cause syntax errors etc. And compiling stopped again to line 156 to error 2. But this is not problem. I have working Stellarium from previous compiling.

[ 77%] Built target LensDistortionEstimator-static_autogen_timestamp_deps
[ 77%] Built target TelescopeControl_NexStar_autogen
[ 77%] Built target ShowMySky_autogen_timestamp_deps
[ 77%] Built target TelescopeControl_Lx200
[ 77%] Built target TelescopeControl_Rts2
[ 77%] Built target TelescopeControl_INDI_autogen
[ 77%] Built target ShowMySky_autogen
[ 77%] Built target TelescopeControl_NexStar
[ 77%] Linking CXX shared library libShowMySky-Qt6.so
[ 78%] Built target TelescopeControl_INDI
[ 78%] Built target TelescopeControl_gui_autogen_timestamp_deps
[ 78%] Automatic MOC and UIC for target LensDistortionEstimator-static
[ 78%] Built target TelescopeControl_gui_autogen
[ 80%] Built target TelescopeControl_gui
[ 80%] Built target TelescopeControl-static_autogen_timestamp_deps
/usr/bin/ld: ../libcommon.a(Spectrum.cpp.o): warning: relocation against `_ZTISt12out_of_range@@GLIBCXX_3.4' in read-only section `.text.unlikely'
[ 80%] Automatic MOC and UIC for target TelescopeControl-static
/usr/bin/ld: ../libcommon.a(EclipsedDoubleScatteringPrecomputer.cpp.o): relocation R_X86_64_PC32 against symbol `_ZTISt12out_of_range@@GLIBCXX_3.4' can not be used when making a shared object; käännetty uudelleen -fPIC-parametrillä
/usr/bin/ld: loppulinkitys ei onnistunut: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [_deps/showmysky-qt6-build/ShowMySky/CMakeFiles/ShowMySky.dir/build.make:160: _deps/showmysky-qt6-build/ShowMySky/libShowMySky-Qt6.so.15.0.0] Virhe 1
make[1]: *** [CMakeFiles/Makefile2:3376: _deps/showmysky-qt6-build/ShowMySky/CMakeFiles/ShowMySky.dir/all] Virhe 2
make[1]: *** Odotetaan keskeneräisiä töitä....
[ 80%] Built target TelescopeControl-static_autogen
[ 80%] Built target LensDistortionEstimator-static_autogen
make: *** [Makefile:156: all] Virhe 2

**Reply by 10110111** [reply]
> Command diff does not have --git parameter

It's not a command to run, it's a patch to apply. You can apply it by e.g. running the following command:
```bash
patch -Np1 <<'EOF'
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac653d5caa..37fd0327c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -691,8 +691,7 @@ SET(CMAKE_INCLUDE_CURRENT_DIR ON)
 IF(ENABLE_SHOWMYSKY)
     ADD_DEFINITIONS(-DENABLE_SHOWMYSKY)
     CPMFindPackage(NAME ShowMySky-Qt${QT_VERSION_MAJOR}
-        URL https://github.com/10110111/CalcMySky/archive/refs/tags/v0.3.3.tar.gz
-        URL_HASH SHA256=21cce3187009ce62a4a08a72b4e22ae7bc00654edfbc7fa9fe8ab65b412791df
+        URL https://codeload.github.com/10110111/CalcMySky/zip/refs/heads/master
         EXCLUDE_FROM_ALL yes
         OPTIONS "QT_VERSION ${QT_VERSION_MAJOR}")
     IF(ShowMySky-Qt${QT_VERSION_MAJOR}_ADDED)
EOF
```

**Comment by alex-w** [comment]
`-Who-dev` - what is it?

**Reply by ilkant** [reply]
make[1]: *** Odotetaan keskeneräisiä töitä....
In file included from /home/ilpo/temp/stellarium/src/core/modules/Landscape.hpp:25,
                 from /home/ilpo/temp/stellarium/build/unix/plugins/Scenery3d/src/Scenery3d-static_autogen/EWIEGA46WW/../../../../../../../plugins/Scenery3d/src/S3DRenderer.hpp:26,
                 from /home/ilpo/temp/stellarium/build/unix/plugins/Scenery3d/src/Scenery3d-static_autogen/EWIEGA46WW/moc_S3DRenderer.cpp:9,
                 from /home/ilpo/temp/stellarium/build/unix/plugins/Scenery3d/src/Scenery3d-static_autogen/mocs_compilation.cpp:3:
/home/ilpo/temp/stellarium/src/core/StelFader.hpp:104:20: warning: ‘virtual StelFader& LinearFader::operator=(bool)’ was hidden [-Woverloaded-virtual=]
  104 |         StelFader& operator=(bool s) override
      |                    ^~~~~~~~
/home/ilpo/temp/stellarium/src/core/StelFader.hpp:139:7: note:   by ‘ParabolicFader::operator=’
  139 | class ParabolicFader : public LinearFader
      |       ^~~~~~~~~~~~~~
In file included from /home/ilpo/temp/stellarium/src/core/modules/StarMgr.hpp:26,
                 from /home/ilpo/temp/stellarium/plugins/LensDistortionEstimator/src/gui/LensDistortionEstimatorDialog.cpp:22:
/home/ilpo/temp/stellarium/src/core/StelFader.hpp:104:20: warning: ‘virtual StelFader& LinearFader::operator=(bool)’ was hidden [-Woverloaded-virtual=]
  104 |         StelFader& operator=(bool s) override
      |                    ^~~~~~~~
/home/ilpo/temp/stellarium/src/core/StelFader.hpp:139:7: note:   by ‘ParabolicFader::operator=’
  139 | class ParabolicFader : public LinearFader
      |       ^~~~~~~~~~~~~~
make: *** [Makefile:156: all] Virhe 2

**Reply by alex-w** [reply]
Sorry, but maybe `-Wno-dev`?

Hint: `CMAKE_SUPPRESS_DEVELOPER_WARNINGS`

**Reply by ilkant** [reply]
$ make -j12 -Wno-dev
[  0%] Built target MissingStars-static_autogen_timestamp_deps
[  0%] Built target Novae-static_autogen_timestamp_deps
[  0%] Built target Exoplanets-static_autogen_timestamp_deps
[  0%] Built target Calendars-static_autogen_timestamp_deps
[  0%] Built target MeteorShowers-static_autogen_timestamp_deps
[  0%] Built target NavStars-static_autogen_timestamp_deps
[  0%] Built target AngleMeasure-static_autogen_timestamp_deps
[  0%] Built target TelescopeControl_common_autogen_timestamp_deps
[  0%] Built target EquationOfTime-static_autogen_timestamp_deps
[  0%] Built target indiclient_autogen_timestamp_deps
[  0%] Built target ArchaeoLines-static_autogen_timestamp_deps
[  0%] Built target Observability-static_autogen_timestamp_deps
[  0%] Built target OnlineQueries-static_autogen_timestamp_deps
[  0%] Built target Quasars-static_autogen_timestamp_deps
[  0%] Built target Oculars-static_autogen_timestamp_deps
[  0%] Built target PointerCoordinates-static_autogen_timestamp_deps
[  0%] Built target Satellites-static_autogen_timestamp_deps
[  0%] Built target Pulsars-static_autogen_timestamp_deps
[  0%] Built target Scenery3d-static_autogen_timestamp_deps
[  0%] Built target SolarSystemEditor-static_autogen_timestamp_deps
[  0%] Built target RemoteSync-static_autogen_timestamp_deps
[  0%] Built target RemoteControl-static_autogen_timestamp_deps
[  0%] Built target generate-cpp
[  0%] Built target Supernovae-static_autogen_timestamp_deps
[  0%] Generating check-version-update
[  0%] Built target QXlsx_autogen_timestamp_deps
[  0%] Built target common_autogen_timestamp_deps
[  0%] Built target glues_stel_autogen_timestamp_deps
[  0%] Built target qtcompress_stel_autogen_timestamp_deps
[  0%] Built target TextUserInterface-static_autogen_timestamp_deps
[  6%] Built target translations-stellarium-skycultures
[ 12%] Built target translations-stellarium
[ 12%] Built target splash
[ 13%] Generating version.cpp
[ 20%] Built target translations-stellarium-planetary-features
[ 20%] Built target icon
[ 26%] Built target translations-stellarium-scripts
[ 32%] Built target translations-stellarium-remotecontrol
[ 32%] Built target TelescopeControl_common_autogen
[ 32%] Automatic MOC and UIC for target indiclient
[ 32%] Built target version_autogen_timestamp_deps
[ 32%] Built target AngleMeasure-static_autogen
[ 32%] Built target EquationOfTime-static_autogen
[ 32%] Built target ArchaeoLines-static_autogen
[ 32%] Built target MeteorShowers-static_autogen
[ 32%] Built target Calendars-static_autogen
[ 32%] Built target MissingStars-static_autogen
[ 32%] Built target NavStars-static_autogen
[ 32%] Built target indiclient_autogen
[ 32%] Built target Novae-static_autogen
[ 33%] Built target Observability-static_autogen
[ 34%] Built target PointerCoordinates-static_autogen
[ 35%] Built target Quasars-static_autogen
[ 35%] Built target Oculars-static_autogen
[ 35%] Built target Pulsars-static_autogen
[ 35%] Built target ManPages
[ 35%] Automatic MOC and UIC for target Exoplanets-static
[ 35%] Built target RemoteSync-static_autogen
[ 35%] Automatic MOC and UIC for target OnlineQueries-static
[ 35%] Built target RemoteControl-static_autogen
[ 35%] Built target nlopt_autogen_timestamp_deps
[ 35%] Built target Satellites-static_autogen
[ 35%] Built target SolarSystemEditor-static_autogen
[ 35%] Built target Supernovae-static_autogen
[ 35%] Built target TextUserInterface-static_autogen
[ 35%] Built target glues_stel_autogen
[ 35%] Built target qtcompress_stel_autogen
[ 35%] Built target TelescopeControl_common
[ 35%] Built target AngleMeasure-static
[ 35%] Built target Exoplanets-static_autogen
[ 36%] Built target ArchaeoLines-static
[ 37%] Built target EquationOfTime-static
[ 42%] Built target Calendars-static
[ 43%] Built target MeteorShowers-static
[ 44%] Built target MissingStars-static
[ 45%] Built target NavStars-static
[ 45%] Built target Observability-static
[ 46%] Built target Novae-static
[ 46%] Built target PointerCoordinates-static
[ 47%] Built target Oculars-static
[ 47%] Built target Quasars-static
[ 47%] Built target Pulsars-static
[ 49%] Built target RemoteSync-static
[ 50%] Built target SolarSystemEditor-static
[ 53%] Built target Supernovae-static
[ 53%] Built target RemoteControl-static
[ 55%] Built target Satellites-static
[ 56%] Built target glues_stel
[ 58%] Built target TextUserInterface-static
[ 58%] Built target qtcompress_stel
[ 58%] Built target TelescopeControl_Lx200_autogen_timestamp_deps
[ 58%] Built target TelescopeControl_NexStar_autogen_timestamp_deps
[ 58%] Built target TelescopeControl_Rts2_autogen_timestamp_deps
[ 58%] Built target TelescopeControl_NexStar_autogen
[ 60%] Built target TelescopeControl_Lx200_autogen
[ 61%] Built target TelescopeControl_Rts2_autogen
[ 61%] Built target TelescopeControl_NexStar
[ 61%] Built target TelescopeControl_Lx200
[ 61%] Built target TelescopeControl_Rts2
[ 61%] Automatic MOC and UIC for target version
[ 61%] Automatic MOC and UIC for target common
[ 62%] Automatic MOC and UIC for target nlopt
[ 63%] Built target indiclient
[ 63%] Built target OnlineQueries-static_autogen
[ 63%] Automatic MOC and UIC for target QXlsx
[ 64%] Built target Exoplanets-static
[ 64%] Automatic MOC and UIC for target Scenery3d-static
[ 64%] Built target TelescopeControl_INDI_autogen_timestamp_deps
[ 64%] Built target common_autogen
[ 64%] Built target nlopt_autogen
[ 64%] Built target version_autogen
[ 64%] Built target QXlsx_autogen
[ 64%] Generating check-version-update
[ 64%] Generating version.cpp
[ 65%] Built target OnlineQueries-static
[ 65%] Built target Scenery3d-static_autogen
[ 69%] Built target nlopt
[ 69%] Automatic MOC and UIC for target TelescopeControl_INDI
[ 70%] Built target common
[ 70%] Generating check-version-update
[ 70%] Generating version.cpp
[ 70%] Built target LensDistortionEstimator-static_autogen_timestamp_deps
[ 70%] Built target TelescopeControl_INDI_autogen
[ 71%] Built target version
[ 75%] Built target QXlsx
[ 77%] Built target Scenery3d-static
[ 77%] Built target ShowMySky_autogen_timestamp_deps
[ 77%] Built target ShowMySky_autogen
[ 78%] Built target TelescopeControl_INDI
[ 78%] Automatic MOC and UIC for target LensDistortionEstimator-static
[ 78%] Built target TelescopeControl_gui_autogen_timestamp_deps
[ 78%] Linking CXX shared library libShowMySky-Qt6.so
[ 78%] Built target LensDistortionEstimator-static_autogen
[ 78%] Built target TelescopeControl_gui_autogen
[ 80%] Built target TelescopeControl_gui
[ 81%] Built target LensDistortionEstimator-static
[ 81%] Built target TelescopeControl-static_autogen_timestamp_deps
[ 81%] Built target TelescopeControl-static_autogen
[ 82%] Built target TelescopeControl-static
/usr/bin/ld: ../libcommon.a(Spectrum.cpp.o): warning: relocation against `_ZTISt12out_of_range@@GLIBCXX_3.4' in read-only section `.text.unlikely'
[ 82%] Built target AllStaticPlugins
/usr/bin/ld: ../libcommon.a(EclipsedDoubleScatteringPrecomputer.cpp.o): relocation R_X86_64_PC32 against symbol `_ZTISt12out_of_range@@GLIBCXX_3.4' can not be used when making a shared object; käännetty uudelleen -fPIC-parametrillä
/usr/bin/ld: loppulinkitys ei onnistunut: bad value
[ 82%] Built target stelMain_autogen_timestamp_deps
collect2: error: ld returned 1 exit status
make[2]: *** [_deps/showmysky-qt6-build/ShowMySky/CMakeFiles/ShowMySky.dir/build.make:160: _deps/showmysky-qt6-build/ShowMySky/libShowMySky-Qt6.so.15.0.0] Virhe 1
make[1]: *** [CMakeFiles/Makefile2:3376: _deps/showmysky-qt6-build/ShowMySky/CMakeFiles/ShowMySky.dir/all] Virhe 2
make[1]: *** Odotetaan keskeneräisiä töitä....
[ 82%] Automatic MOC and UIC for target stelMain
[ 82%] Built target stelMain_autogen
make: *** [Makefile:156: all] Virhe 2

**Comment by ilkant** [comment]
(Kubuntu 24.10)
I patched Make CMakeLists.txt , but again:

---
[ 42%] Building CXX object plugins/Calendars/src/CMakeFiles/Calendars-static.dir/VietnameseCalendar.cpp.o
[ 42%] Linking CXX static library libCalendars.a
[ 42%] Built target Calendars-static
make: *** [Makefile:156: all] Virhe 2
$

**Reply by 10110111** [reply]
You didn't show the actual error.

**Comment by ilkant** [comment]
make: *** [Makefile:156: all] Error 2

$(MAKE) $(MAKESILENT) -f CMakeFiles/Makefile2 all                  <- Makefile line 156

**Reply by ilkant** [reply]
Today I deleted all directories of Stellarium compile and then I downloaded Github codes again and compiled. No errors. And Stellarium started, but the screen was zoomed and it didn't show menus etc. I try to find where I can set screen parameters better. My OS uses Qt 6. I restore old configuration files of Stellarium from backups. Maybe screen works fine. Now screen works fine. All problems solved. Thank you.

Compile error was on old sources. Or some lines of old confituration files.
